2022/Q3/artifacts/PoS-tla/Staking.tla

```diff
@@ -205,14 +220,27 @@ ComputedUnbonds(setBonds, totalAmount, e) == LET
                                          ELSE 
                                           LET min == Min(record.remaining, bond.amount) 
                                           IN [ remaining |-> record.remaining - min,
-                                               unbonds |-> record.unbonds \union {[id |-> record.id, amount |-> min, start |-> bond.epoch, end |-> e]},
-                                               bonds |-> record.bonds \union {bond},
-                                               modifiedBond |-> 
+                                               unbonds |-> record.unbonds \union {[id |-> record.id, amount |-> min, start |-> bond.start, end |-> e]},
+                                               bondsToRemove |-> record.bondsToRemove \union {bond},
+                                               bondsToAdd |-> record.bondsToAdd \union {[id |-> bond.id, amount |-> min, start |-> bond.start, end |-> epoch + PipelineLength]} \union
                                                 IF bond.amount = min
                                                 THEN {}
-                                                ELSE {[id |-> idBonds, amount |-> bond.amount - min, start |-> bond.epoch]},
+                                                ELSE {[id |-> idBonds, amount |-> bond.amount - min, start |-> bond.start, end |-> -1]},
                                                id |-> record.id + 1 ]
-                                        IN ApaFoldSet(F, [ remaining |-> totalAmount, unbonds |-> {}, bonds |-> {}, modifiedBond |-> {}, id |-> idUnbonds], setBonds)
+                                        IN ApaFoldSet(F, [ remaining |-> totalAmount, unbonds |-> {}, bondsToRemove |-> {}, bondsToAdd |-> {}, id |-> idUnbonds], setBonds)
+
+\* @type: (Int, Set(SLASH)) => Int;
+BondAfterSlashing(amount, setSlashes) == LET
+                                         \* @type: (Int, SLASH) => Int;
+                                         F(total, slash) == total + (amount - amount*slash.finalRate)
+                                         IN ApaFoldSet(F, 0, setSlashes)
+
+\* @type: (ADDR, Set(BOND)) => Int;
+ComputeTakeTotalDeltas(val, setBonds) == LET
```

Comment:
I think I understand what this is doing, but I get lost when trying to related to the name (`ComputeTakeTotalDeltas`)

Reply:
Agree, it is a bad name. Any suggestion?